src/main/download.py

```diff
@@ -1,31 +1,36 @@
-from pathlib import Path
-import requests
 import os
 
-from transformers import AutoTokenizer, AutoModelForQuestionAnswering
+from transformers import AutoTokenizer, AutoModelForQuestionAnswering, AutoModel
 
 
-def download_model(path, model):
+def download_model(path, model_name):
     """
     Downloads models from Hugging Face is the 'download' flag has been set to true.
     If a folder with the model name already exists, then takes no action.
     Otherwise, the model should exist on disk.
     @param path: Path to where the models will be downloaded.
-    @param model: Model object.
-    @return: Path to the model on disk.
+    @param model_name: Model name.
+    @return: Path to the model on disk and if the model already exists
+    @raise: throws an error if the model can't be downloaded and remove dir
     """
-    model_name = model['model']
     model_path = "{}/{}/".format(path, model_name)
-    if not os.path.isdir(model_path):
-        os.makedirs(model_path)  # Create the dir
-        print("Downloading model {} to {}".format(model_name, model_path))
-        download_from_huggingface(model_name, model_path)
-    else:
-        print("Model already exists: {}".format(model_path))
-    return model_path
+    model_exists = False
+    try:
+        if not os.path.isdir(model_path):
+            os.makedirs(model_path)  # Create the dir
+            print("Downloading model {} to {}".format(model_name, model_path))
+            download_from_huggingface(model_name, model_path)
+        else:
+            print("Model already exists: {}".format(model_path))
+            model_exists = True
+    except Exception as e:
+        os.rmdir(model_path)
+        raise e
 
+    return model_path, model_exists
 
-def download_from_huggingface(model_name, model_path):
+
+def download_from_huggingface_qa(model_name, model_path):
```

Comment:
I recommend changing the function name to `download_from_huggingface_question_answers`. I know it is long, but I prefer descriptive names.